sw/airborne/firmwares/rotorcraft/stabilization/stabilization_indi.c

```diff
@@ -334,85 +359,102 @@ static void stabilization_indi_calc_cmd(struct Int32Quat *att_err, bool rate_con
 
   g2_times_du = 0.0;
   int8_t i;
-  for(i=0; i<INDI_NUM_ACT; i++) {
-    g2_times_du += g2[i]*indi_du[i];
+  for (i = 0; i < INDI_NUM_ACT; i++) {
+    g2_times_du += g2[i] * indi_du[i];
   }
   //G2 is scaled by INDI_G_SCALING to make it readable
-  g2_times_du = g2_times_du/INDI_G_SCALING;
-
-  // Calculate the increment for each actuator
-  for(i=0; i<INDI_NUM_ACT; i++) {
-    indi_du[i] = (g1g2_pseudo_inv[i][0] * (angular_accel_ref.p - angular_acceleration[0]))
-               + (g1g2_pseudo_inv[i][1] * (angular_accel_ref.q - angular_acceleration[1]))
-               + (g1g2_pseudo_inv[i][2] * (angular_accel_ref.r - angular_acceleration[2] + g2_times_du));
-  }
-
-  if(indi_thrust_increment_set){
-    // The required body-z acceleration is calculated by the outer loop INDI controller
-    for(i=0; i<INDI_NUM_ACT; i++) {
-      indi_du[i] = indi_du[i] + g1g2_pseudo_inv[i][3]*indi_thrust_increment;
+  g2_times_du = g2_times_du / INDI_G_SCALING;
+
+  float v_thrust = 0.0;
+  if (indi_thrust_increment_set) {
+    v_thrust = indi_thrust_increment;
+
+    //update thrust command such that the current is correctly estimated
+    stabilization_cmd[COMMAND_THRUST] = 0;
+    int32_t num_thrusters = INDI_NUM_ACT;
+    for (i = 0; i < INDI_NUM_ACT; i++) {
+      stabilization_cmd[COMMAND_THRUST] += actuator_state[i] * -((int32_t) act_is_servo[i] - 1);
+      // Assume all non-servos are delivering thrust
+      num_thrusters -= act_is_servo[i];
     }
+    stabilization_cmd[COMMAND_THRUST] /= num_thrusters;
```

Comment:
I guess the num_thrusters is a fixed number (i.e. preconfigured) and can be computed in the init.